avocado/utils/astring.py

```diff
@@ -326,4 +326,9 @@ def to_text(data, encoding=ENCODING):
         if encoding is None:
             encoding = ENCODING
         return data.decode(encoding)
+    elif not isinstance(data, string_types):
+        if sys.version_info[0] < 3:
+            return unicode(data)    # pylint: disable=E0602
```

Comment:
OK, the logic seems simple enough, and adds value.  I'm a interested in learning more about two things:

* the `unicode()` behavior on error scenarios
* the fact that `encoding` is not passed to it

I'll write some experiments/tests and get back to this.  If you have any comments, please let me know.

Reply:
Hard to say. I played with it for a while, when 4bytes unicode is enabled it seems to work well even with utf-16 chars without specifying anything else so I just used that. Let me know of your findings, alternatively we can improve that when we get user reports...